index.src.html

```diff
@@ -1380,6 +1380,28 @@ <h4 id="body-extraction" algorithm>
           6.  Set |Content-Type| to
               `application/x-www-form-urlencoded;charset=UTF-8`.
 
+          Otherwise, |data| is a generic object, so:
+
+          1.  If |data| has a property named like the |credential|'s
+              {{PasswordCredential/idName}} or
+              {{PasswordCredential/passwordName}}, <a>throw</a> a `TypeError`
+              exception.
+
+          2.  Add a property to |data| with name being the |credential|'s
+              {{PasswordCredential/idName}} and value the |credential|'s
+              {{Credential/id}}.
+
+          3.  Add a property to |data| with name being the |credential|'s
+              {{PasswordCredential/passwordName}} and value the |credential|'s
+              {{[[password]]}}.
+
+          4.  Push the result of applying `JSON.stringify` to |data| with
+              "`utf-8`" as the explicit character encoding, to |body|'s <a
+              for="body">stream</a>.
```

Comment:
> Looking at https://fetch.spec.whatwg.org/#concept-readablestream I am not sure what "push" means. The nearest I see there is "enqueue". I am not sure how to specify the encoding for that operation either. I'm afraid I just blindly copied the pattern of the other branches for FormData and URLSearchParams.

Yeah, it looks like all of this has moved out from under this spec. I think you'll want to do something like "Set |action| to an action that executes {{JSON.stringify()}} on |data|." Fetch will take care of the rest (because we're defining the switch part in step 4 of https://fetch.spec.whatwg.org/#concept-bodyinit-extract, and |action| is poked at in step 6).